crates/module-system/sov-modules-api/src/gas/traits.rs

```diff
@@ -570,6 +570,15 @@ pub trait GasMeter {
         Ok(())
     }
 
+    /// Returns the amount of gas available. Used to set the EVM gas limit
+    fn gas_limit(
+        &mut self,
+    ) -> Result<<Self::Spec as Spec>::Gas, GasMeteringError<<Self::Spec as Spec>::Gas>> {
+        unreachable!(
+            "Default implementation should not be called. Override in the respective gas meter"
```

Comment:
We can remove the default implementation of the method, which will require the trait implementer to provide their own.